lua/tardis/cl_hud.lua

```diff
@@ -2,41 +2,74 @@
 
 surface.CreateFont("TARDIS-HUD-Large", {
     font="Roboto",
-    size=68
+    size=58
 })
 
 surface.CreateFont("TARDIS-HUD-Med", {
     font="Roboto",
-    size=40
+    size=36
 })
 
 surface.CreateFont("TARDIS-HUD-Small", {
     font="Roboto",
-    size=34
+    size=30
 })
 
+local fgcolor = NamedColor("FgColor")
+local red = NamedColor("Caution")
+local bgcolor = NamedColor("BgColor")
+
+local function CreatePercentageHUDPanel(text, value, offset, red_level)
+    local width = 115
+    local height = (ScrW() >= 800) and 95 or 85
+
+    if value >= 10 then width = width + 10 end
+    if value == 100 then width = width + 20 end
+
+    local x = (ScrW() - width) * 0.02 + offset
+    local y = (ScrH() - height) * 0.025
+
+    local value_font = (height == 95) and "TARDIS-HUD-Large" or "TARDIS-HUD-Med"
+    local textcolor = (value > red_level) and fgcolor or red
+
+    draw.RoundedBox( 10, x, y, width, height, bgcolor )
+    draw.DrawText( text, "TARDIS-HUD-Small", x+10, y+10, textcolor, TEXT_ALIGN_LEFT )
+    draw.DrawText( tostring(value) .. "%", value_font, x+10, y+35, textcolor, TEXT_ALIGN_LEFT )
+
+    return width, height
+end
+
 hook.Add("HUDPaint", "TARDIS-HUD", function()
-    if not TARDIS:GetSetting("health-enabled") then return end
-    if not (LocalPlayer():GetTardisData("interior") or LocalPlayer():GetTardisData("exterior")) then return end
-    local tardis = LocalPlayer():GetTardisData("exterior")
+    local ply = LocalPlayer()
+    if ( GetConVarNumber( "cl_drawhud" ) == 0 ) then return end
+    local tardis = ply:GetTardisData("exterior")
+    if not tardis then return end
+
+    local draw_health = TARDIS:GetSetting("health-enabled")
+    local draw_artron = TARDIS:GetSetting("artron_energy")
+
+    if not draw_health and not draw_artron then return end
+
     if not IsValid(tardis) then return end
```

Comment:
We are checking the TARDIS above on line 46 if it exists and then checking if it IsValid down here, we should move this up and replace the check on line 46

Reply:
Fixed 226ea43a4eafd4c6e1bab819a3bd07fb890b3a56